autopush/endpoint.py

```diff
@@ -220,3 +256,115 @@ def write_error(self, code, exception=None):
         if exception is not None:
             log.err(exception)
         self.finish()
+
+
+class RegistrationHandler(cyclone.web.RequestHandler):
+
+    def _error_response(self, failure):
+        log.err(failure)
+        self.set_status(500)
+        self.write("Error processing request")
+        self.finish()
+
+    def _load_params(self):
+        tags = {'chid': 'channelid',
+                'conn': 'connect',
+                }
+        chid = conn = None
+        if len(self.request.body) > 0:
```

Comment:
I dunno, there's a lot in common, but subtle differences (e.g. Endpoint does int conversion and checking on "version") Not saying that it's not impossible to build a generic handler, but I think we'd be introducing more complexity than it's probably worth.